core/src/commonMain/kotlin/com/powersync/utils/Json.kt

```diff
@@ -12,3 +17,32 @@ internal object JsonUtil {
     }
 }
 
+internal fun convertMapToJson(map: Map<String, Any?>?): JsonObject {
+    if (map == null) return JsonObject(emptyMap())
+
+    val result = map.mapValues { (_, value) ->
```

Comment:
Is there custom mapping necessary? Would `JsonObject(map)` not convert it to a `JsonObject` with the correct types? Does it fail to convert some types to string in the `JsonObject`?

Reply:
`JsonObject` requires a `Map<String, JsonElement>` while we are receiving a `Map<String, Any?>` so we need to convert the values to a JsonElement before we can run `JsonObject` on the map, unless I am missing something?